tests/account_change_password_test.go

```diff
@@ -100,6 +101,83 @@ func TestChangePassword(t *testing.T) {
 			StatusCode: 200,
 		})
 	})
+
+	// sytest: Pushers created with a different access token are deleted on password change
+	t.Run("Pushers created with a different access token are deleted on password change", func(t *testing.T) {
+		sessionOptional := createSession(t, deployment, "test_change_password_user", "new_optional_password")
+		reqBody := client.WithJSONBody(t, map[string]interface{}{
+			"data": map[string]interface{}{
+				"url": "https://dummy.url/_matrix/push/v1/notify",
+			},
+			"profile_tag":         "tag",
+			"kind":                "http",
+			"app_id":              "complement",
+			"app_display_name":    "complement_display_name",
+			"device_display_name": "device_display_name",
+			"pushkey":             "a_push_key",
+			"lang":                "en",
+		})
+
+		res := sessionOptional.DoFunc(t, "POST", []string{"_matrix", "client", "r0", "pushers", "set"}, reqBody)
+
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: 200,
+		})
+
+		changePassword(t, passwordClient, "new_optional_password", newPassword)
+
+		res = passwordClient.DoFunc(t, "GET", []string{"_matrix", "client", "r0", "pushers"})
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: 200,
+			JSON: []match.JSON{
+				match.JSONKeyPresent("pushers"),
+				match.JSONArrayEach("pushers", func(val gjson.Result) error {
+					if len(val.Array()) != 0 {
+						return fmt.Errorf("expected array length to be zero: %v", val.Raw)
+					}
+					return nil
+				}),
+			},
+		})
+	})
+
+	// sytest: Pushers created with a the same access token are not deleted on password change
+	t.Run("Pushers created with a the same access token are not deleted on password change", func(t *testing.T) {
+		reqBody := client.WithJSONBody(t, map[string]interface{}{
+			"data": map[string]interface{}{
+				"url": "https://dummy.url/_matrix/push/v1/notify",
+			},
+			"profile_tag":         "tag",
+			"kind":                "http",
+			"app_id":              "complement",
+			"app_display_name":    "complement_display_name",
+			"device_display_name": "device_display_name",
+			"pushkey":             "a_push_key",
+			"lang":                "en",
+		})
+
+		res := passwordClient.DoFunc(t, "POST", []string{"_matrix", "client", "r0", "pushers", "set"}, reqBody)
+
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: 200,
+		})
+
+		changePassword(t, passwordClient, newPassword, "new_optional_password")
+
+		res = passwordClient.DoFunc(t, "GET", []string{"_matrix", "client", "r0", "pushers"})
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: 200,
+			JSON: []match.JSON{
+				match.JSONKeyPresent("pushers"),
+				match.JSONArrayEach("pushers", func(val gjson.Result) error {
+					if len(val.Array()) == 1 {
+						return fmt.Errorf("expected array length to be zero: %v", val.Raw)
```

Comment:
Shouldn't this be `if len(val.Array()) != 1` and then `expected array length to be 1` ?

Reply:
@kegsay yes, I fixed it too! But this is the weird error I am getting, 
```
=== CONT  TestChangePassword/Pushers_created_with_a_the_same_access_token_are_not_deleted_on_password_change
52
    account_change_password_test.go:161: MatchResponse expected array length to be one: {"app_display_name":"complement_display_name","app_id":"complement","data":{"url":"https://dummy.url/_matrix/push/v1/notify"},"device_display_name":"device_display_name","kind":"http","lang":"en","profile_tag":"tag","pushkey":"a_push_key"} - http://172.17.0.1:49158/_matrix/client/r0/pushers => {"pushers":[{"app_display_name":"complement_display_name","app_id":"complement","data":{"url":"https://dummy.url/_matrix/push/v1/notify"},"device_display_name":"device_display_name","kind":"http","lang":"en","profile_tag":"tag","pushkey":"a_push_key"}]}
```
As can be seen pusher has array length 1 clearly in the logs.

Reply:
Please do let me know if I missed anything

Reply:
Bump @kegsay

Reply:
Hmm, `pushers` is the array, and presumably you just want to check the array size. In which case you don't need `JSONArrayEach` at all, as that function will loop each element of the array.

I would just increment a number on `JSONArrayEach` and then check if that number is 0 or 1 e.g:
```go
size := 0
match.JSONArrayEach("pushers", func(val gjson.Result) error {
    size += 1
    return nil
})
if size != 1 {
    t.Errorf(...)
}
```